lib/Optimizer/CodeGen/LowerToQIR.cpp

```diff
@@ -1214,6 +1260,47 @@ class StdvecSizeOpPattern
   }
 };
 
+class CreateStringLiteralOpPattern
+    : public ConvertOpToLLVMPattern<cudaq::cc::CreateStringLiteralOp> {
+public:
+  using Base = ConvertOpToLLVMPattern<cudaq::cc::CreateStringLiteralOp>;
+  using Base::Base;
+
+  LogicalResult
+  matchAndRewrite(cudaq::cc::CreateStringLiteralOp stringLiteralOp,
+                  OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    auto loc = stringLiteralOp.getLoc();
+    auto parentModule = stringLiteralOp->getParentOfType<ModuleOp>();
+    auto context = parentModule->getContext();
+
+    StringRef stringLiteral = stringLiteralOp.getStringLiteral();
+
+    // Write to the module body
+    auto insertPoint = rewriter.saveInsertionPoint();
+    rewriter.setInsertionPointToStart(parentModule.getBody());
+
+    // Create the register name global
+    auto builder = cudaq::IRBuilder::atBlockEnd(parentModule.getBody());
+    auto slGlobal =
+        builder.genCStringLiteralAppendNul(loc, parentModule, stringLiteral);
```

Comment:
Cool. I learned something. Back in the day with FIR, I had to roll my own method for this. :)

Reply:
I copied this from code you had written 😄

Reply:
I promptly forgot about this it appears. 😄